addon/validations/factory.js

```diff
@@ -396,6 +396,14 @@ function createTopLevelPropsMixin(validatableAttrs) {
       return get(this, 'messages.0');
     })).readOnly(),
 
+    warnings: computed(...validatableAttrs.map(attr => `attrs.${attr}.@each.warning`), function () {
+      return emberArray(flatten(validatableAttrs.map(attr => get(this, `attrs.${attr}.warning`)))).compact();
+    }).readOnly(),
+
+    warning: computed('warning.[]', cycleBreaker(function () {
+      return get(this, 'warning.0');
```

Comment:
This is the same code that is the `errors` and `error` CPs. I believe I tried firstObject but it errored on me. Ill give it another shot though.